mysys/crc32/crc32_arm64.c

```diff
@@ -131,109 +139,122 @@ uint32_t crc32c_aarch64(uint32_t crc, const unsigned char *buffer, uint64_t len)
 
 	if (buffer) {
 
-/* Crypto extension Support
- * Process 1024 Bytes (per block)
- */
+		/* Pmull runtime check here.
+		 * Raspberry Pi 4 supports crc32 but doesn't support pmull (MDEV-23030).
+		 *
+		 * Consider the condition that the target platform does support hardware crc32
+		 * but not support PMULL. In this condition, it should leverage the aarch64
+		 * crc32 instruction (__crc32c) and just only skip parallel computation (pmull/vmull)
+		 * rather than skip all hardware crc32 instruction of computation.
+		 */
+		if (pmull_aarch64_runtime_flag) {
+
+/* The following Macro (HAVE_ARMV8_CRYPTO) is used for compiling check */
 #ifdef HAVE_ARMV8_CRYPTO
 
-/* Intrinsics Support  */
+/* Crypto extension Support
+ * Parallel computation with 1024 Bytes (per block)
+ * Intrinsics Support
+ */
 #ifdef HAVE_ARMV8_CRC_CRYPTO_INTRINSICS
-		const poly64_t k1 = 0xe417f38a, k2 = 0x8f158014;
-		uint64_t t0, t1;
```

Comment:
This code is rather large. Could we split each implementation into a `static` function? The compiler should be able inline those.

I would prefer to indent preprocessor directives too:
```c
#ifdef HAVE_ARMV8_CRYPTO
# ifdef HAVE_ARMV8_CRC_CRYPTO_INTRINSICS
```
Also, I would suggest to try to follow the common code formatting style, given that the entire function would be reformatted by this patch. Only InnoDB historically used TAB for indentation.